### PR TITLE
ref(derive_code_mappings): Use logging instead of capture message

### PR DESCRIPTION
**Pull request by armenzg**

### CODE REVIEW

File: src/sentry/tasks/derive_code_mappings.py [hunk 1]
```diff
@@ -168,15 +167,10 @@ def report_project_codemappings(
...
     existing_code_mappings = RepositoryProjectPathConfig.objects.filter(project=project)
     if existing_code_mappings.exists():
         msg = f"Project {project.slug}: found {existing_code_mappings=} while attempting to create {code_mappings=} for {stacktrace_paths=}"
-        sentry_msg = "derive_code_mappings: code mappings already exist."
-    logger.info(msg)
-    logger.info("Reported to Sentry; Query for 'level:info derive'")
-    # Using a generic message will group all events together
-    sentry_sdk.capture_message(sentry_msg)
+
+    logger.info(msg, extra={"org": project.organization.slug, "project": project.slug})
```
Comment (snigdhas) [inline src/sentry/tasks/derive_code_mappings.py:176]: should we give these messages a common prefix? makes it easier to search through gcp logs

**Review by wedamija — APPROVED** [review]
Looks fine to me, although since the message is dynamic it might be a little harder to query in gcp

It could make sense to have the message be consistent and just include the specific data in extra. Up to you though

File: src/sentry/tasks/derive_code_mappings.py (hunk 1 above)
Comment (armenzg) [reply src/sentry/tasks/derive_code_mappings.py:176]: We set the prefix with `logger = logging.getLogger("sentry.tasks.derive_code_mappings")`

**Comment by armenzg** [comment]
I'm convinced. I made the messages generic.

File: src/sentry/tasks/derive_code_mappings.py [hunk 2]
```diff
@@ -167,16 +166,19 @@ def report_project_codemappings(
...
+        msg = "derive_code_mappings: code mappings would have been created."
     existing_code_mappings = RepositoryProjectPathConfig.objects.filter(project=project)
     if existing_code_mappings.exists():
-        msg = f"Project {project.slug}: found {existing_code_mappings=} while attempting to create {code_mappings=} for {stacktrace_paths=}"
-        sentry_msg = "derive_code_mappings: code mappings already exist."
-    logger.info(msg)
-    logger.info("Reported to Sentry; Query for 'level:info derive'")
-    # Using a generic message will group all events together
-    sentry_sdk.capture_message(sentry_msg)
+        msg = "derive_code_mappings: code mappings already exist."
```
Comment (snigdhas) [inline src/sentry/tasks/derive_code_mappings.py:174]: Do we still want to log the existing code mappings for comparison?

Comment (armenzg) [reply src/sentry/tasks/derive_code_mappings.py:180]: Added it now. Good catch!

**Review by snigdhas — APPROVED** [review]

**Comment by snigdhas** [comment]
We need to fix the `test_derive_code_mappings_dry_run` test here since we aren't capturing messages anymore.

**Review by wedamija — APPROVED** [review]

**Pull request merged**